src/components/App.vue

```diff
@@ -453,11 +453,32 @@ export default defineComponent({
       return 'primary';
     });
 
+    const saveDialog = ref(false);
+
+    const saveUrl = urlParams.save as string;
+
+    const handleSave = async () => {
+      if (saveUrl) {
+        const blob = await serialize();
+        fetch(saveUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/zip',
+            'Content-Length': blob.size.toString(),
+          },
+          body: blob,
+        });
```

Comment:
I'd like to handle loading, success, and failure states here. We can show a loading notification when uploading, then a success or failure depending on the state of the fetch.